registry_provider_integration_test.go

```diff
@@ -0,0 +1,575 @@
+//go:build integration
+// +build integration
+
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRegistryProvidersList(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	t.Run("with providers", func(t *testing.T) {
+		orgTest, orgTestCleanup := createOrganization(t, client)
+		defer orgTestCleanup()
+
+		createN := 10
+		providers := make([]*RegistryProvider, 0)
+		// these providers will be destroyed when the org is cleaned up
+		for i := 0; i < createN; i++ {
+			// Create public providers
+			providerTest, _ := createRegistryProvider(t, client, orgTest, PublicRegistry)
+			providers = append(providers, providerTest)
+		}
+		for i := 0; i < createN; i++ {
+			// Create private providers
+			providerTest, _ := createRegistryProvider(t, client, orgTest, PrivateRegistry)
+			providers = append(providers, providerTest)
+		}
+		providerN := len(providers)
+		publicProviderN := createN
+
+		t.Run("returns all providers", func(t *testing.T) {
+			returnedProviders, err := client.RegistryProviders.List(ctx, orgTest.Name, &RegistryProviderListOptions{
+				ListOptions: ListOptions{
+					PageNumber: 0,
+					PageSize:   providerN,
+				},
+			})
+			require.NoError(t, err)
+			assert.NotEmpty(t, returnedProviders.Items)
+			assert.Equal(t, providerN, returnedProviders.TotalCount)
+			assert.Equal(t, 1, returnedProviders.TotalPages)
+			for _, rp := range returnedProviders.Items {
+				foundProvider := false
+				for _, p := range providers {
+					if rp.ID == p.ID {
+						foundProvider = true
+						break
+					}
+				}
+				assert.True(t, foundProvider, "Expected to find provider %s but did not:\nexpected:\n%v\nreturned\n%v", rp.ID, providers, returnedProviders)
+			}
+		})
+
+		t.Run("returns pages", func(t *testing.T) {
```

Comment:
@sebasslash Per this [comment](https://github.com/hashicorp/go-tfe/pull/404#discussion_r876430985), if there isn't enough data to support 999 pages, is that still a valid test? I would guess that we would return pages for data that we have and stop there. 🤔

Reply:
I think it's a simpler check to use `999` for the page number because obviously there aren't 999 pages, so we should expect zero items returned. Which inevitably means the list query params are working as expected.